src/main/kotlin/org/jetbrains/research/ml/coding/assistant/solutionSpace/serialization/TreeContextSerializer.kt

```diff
@@ -0,0 +1,32 @@
+package org.jetbrains.research.ml.coding.assistant.solutionSpace.serialization
+
+import com.github.gumtreediff.io.TreeIoUtils
+import com.github.gumtreediff.tree.TreeContext
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import java.io.ByteArrayOutputStream
+import java.nio.charset.StandardCharsets
+
+object TreeContextSerializer : KSerializer<TreeContext> {
+    override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor("TreeContext", PrimitiveKind.STRING)
+
+    override fun deserialize(decoder: Decoder): TreeContext {
+        val xmlString = decoder.decodeString()
+        /*
+        FIXME: newlines transforms into whitespaces while parsing the xml (XMLEventReader.nextEvent()).
+          label="\n" --> label=" "
```

Comment:
We don't need fix it since we remove whitespace. but it is a great idea to add this comment